File: app/src/main/java/com/ichi2/apisample/AnkiDroidConfig.java
package com.ichi2.plaintextimporter;

import java.util.ArrayList;
import java.util.HashMap;


public final class AnkiDroidConfig {
    // Name of deck which will be created in AnkiDroid
    public static final String DECK_NAME = "ImporterSample";
    // Name of model which will be created in AnkiDroid
    public static final String MODEL_NAME = "com.ichi2.plaintextimporter";
    // Optional space separated list of tags to add to every note
    public static final String TAGS = "AnkiImporter2";
    // List of field names that will be used in AnkiDroid model
    public static final String[] FIELDS = {"Front","Back"};

    // Define two keys which will be used when using legacy ACTION_SEND intent
    public static final String FRONT_SIDE_KEY = FIELDS[0];
    public static final String BACK_SIDE_KEY = FIELDS[1];

    /**
     * Generate the ArrayList<HashMap> example data which will be sent to AnkiDroid
     * @return
     */
    public static ArrayList<HashMap<String, String>> getExampleData() {
        final String[] EXAMPLE_FRONTS = {"front1","front2"};
        final String[] EXAMPLE_BACKS = {"back1", "back2"};

        ArrayList<HashMap<String, String>> data = new ArrayList<>();
        for (int idx = 0; idx < EXAMPLE_FRONTS.length; idx++) {
            HashMap<String, String> hm = new HashMap<>();
            hm.put(FIELDS[0], EXAMPLE_FRONTS[idx]);
            hm.put(FIELDS[1], EXAMPLE_BACKS[idx]);
            data.add(hm);
        }
        return data;
    }
}

File: app/src/main/java/com/ichi2/apisample/AnkiDroidConfig.kt
package com.ichi2.apisample

import java.util.*

object AnkiDroidConfig {
    // Name of deck which will be created in AnkiDroid
    val DECK_NAME = "ImporterSample"
    // Name of model which will be created in AnkiDroid
    const val MODEL_NAME = "com.ichi2.plaintextimporter"
    // Optional space separated list of tags to add to every note
    const val TAGS = "AnkiImporter2"
    // List of field names that will be used in AnkiDroid model
    val FIELDS = arrayOf("Front", "Back")

    // Define two keys which will be used when using legacy ACTION_SEND intent
    val FRONT_SIDE_KEY = FIELDS[0]
    val BACK_SIDE_KEY = FIELDS[1]

    /**
     * Generate the ArrayList<HashMap> example data which will be sent to AnkiDroid
     * @return
    </HashMap> */
    val exampleData: ArrayList<HashMap<String, String>>
        get() {
            val exampleFronts = arrayOf("front1", "front2")
            val exampleBacks = arrayOf("back1", "back2")

            return ArrayList<HashMap<String, String>>().apply {
                for (idx in exampleFronts.indices) {
                    val hm = HashMap<String, String>()
                    hm[FIELDS[0]] = exampleFronts[idx]
                    hm[FIELDS[1]] = exampleBacks[idx]
                    add(hm)
                }
            }
        }
}
